src/others/helpers/aid-request-helpers.ts

```diff
@@ -0,0 +1,78 @@
+import { groupBy, map } from "lodash";
+
+import { AidRequest } from "../contexts/api";
+import { ListItem } from "../components/CollapsibleListItem";
+
+export const sortDates = (a: string, b: string) => {
+  return new Date(a).getTime() - new Date(b).getTime();
+};
+
+export const filterByCategoryIds = (aidRequests: AidRequest[], categoryIds: (string | "*")[]): AidRequest[] => {
+  if (categoryIds.includes("*")) return aidRequests;
+  return aidRequests.filter((aidRequest) => categoryIds.includes(aidRequest.category_id));
+};
+
+type Total = {
+  total: number;
+};
+
+type GroupedByCities = {
+  city_id: number;
+  aidRequests: AidRequest[];
+};
+
+export const groupByCityIdWithTotal = (aidRequests: AidRequest[]): (GroupedByCities & Total)[] => {
+  const groupedByCityId = groupBy(aidRequests, "city_id");
+
+  const groupedByCityIdWithTotal = map(groupedByCityId, (reqs, city_id) => {
+    return { city_id: Number(city_id), total: totalCalculator(reqs), aidRequests: [...reqs] };
```

Comment:
why Number is needed here? isn't city_id already a Number?

Reply:
Idk, maybe it gets converted to string on the way, maybe TS compiler gets confused by something else. Without explicit type conversion there are TS errors.